src/validation/variable.rs

```diff
@@ -78,6 +82,75 @@ fn validate_variable(validator: &mut Validator, variable: &Variable, context: &V
     }
 }
 
+/// Checks if a literal or an expression would yield an overflow and if so reports it.
+fn check_if_overflows(validator: &mut Validator, variable: &Variable, context: &ValidationContext) {
+    let Some(initializer) = &variable.initializer else {
+        return
+    };
+
+    let DataTypeDeclaration::DataTypeReference { referenced_type, .. } = &variable.data_type_declaration else {
+        return
+    };
+
+    let Some(dt) = context.index.find_effective_type_by_name(referenced_type) else {
+        return
+    };
+
+    let Ok(Some(initializer)) = const_evaluator::evaluate(initializer, None, context.index) else {
```

Comment:
a variable's initializers aleardy have been evaluated, they are accessible from the variable's index entry like in line 48-50. the const_evaluator even offers a method that will only give a successfully resolved expression:     const_evaluator::get_resolved_constant_statement(...)

this one will also work for expressions that use other constants (like x + y).